src/NuGet.Clients/NuGet.SolutionRestoreManager/InfoBarService.cs

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Telemetry;
+
+#nullable enable
+
+namespace NuGet.SolutionRestoreManager
+{
+    [Export(typeof(IInfoBarService))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class InfoBarService : IInfoBarService, IVsInfoBarUIEvents
```

Comment:
`InfoBarService` class is tied to vulnerability info bar but the class name doesn't reflect that. How about `VulnerabilitiesInfoBar` ?